### PR TITLE
Add auto theme switching to app

### PR DESCRIPTION
**Pull request by JDMathew**
This PR gets user preference mode from the store, select the correct theme based off the mode and pass it down through provider to all components

- Closes: #157

### CODE REVIEW

File: apps/box/src/app/App.tsx [hunk 1]
```diff
@@ -8,16 +8,23 @@ import { NavContainer } from '../navigation/NavContainer';
...
   }
 }
 
+const fullTheme = {
+  light: fxLightTheme,
+  dark: fxDarkTheme,
+};
+
 export const App = () => {
   const [isDarkTheme] = React.useState<boolean>(true);
```
Comment (keithluchtel) [inline apps/box/src/app/App.tsx:25]: remove this state variable and update the status bar logic below.

Comment (JDMathew) [reply apps/box/src/app/App.tsx:25]: Oh, weird it was removed but I think when I rebased with main it was added back in.

Comment (JDMathew) [reply apps/box/src/app/App.tsx]: Have fixed

File: apps/box/src/app/App.tsx [hunk 2]
```diff
@@ -8,16 +8,27 @@ import { NavContainer } from '../navigation/NavContainer';
...
 }
 
+const fullTheme = {
+  light: fxLightTheme,
+  dark: fxDarkTheme,
+};
+
+const barStyles = {
+  light: 'light-content',
+  dark: 'dark-content',
```
[the remark this answers is not in the document]
Comment (JDMathew) [reply apps/box/src/app/App.tsx:26]: Yes, I completely noobed that. Thanks for picking that up.

**Review by keithluchtel — APPROVED** [review]

**Pull request merged**